tree/ntuple/v7/src/RField.cxx

```diff
@@ -2191,6 +2188,93 @@ void ROOT::Experimental::RArrayField::AcceptVisitor(Detail::RFieldVisitor &visit
    visitor.VisitArrayField(*this);
 }
 
+//------------------------------------------------------------------------------
+// RArrayAsRVecField
+
+ROOT::Experimental::RArrayAsRVecField::RArrayAsRVecField(
+   std::string_view fieldName, std::unique_ptr<ROOT::Experimental::Detail::RFieldBase> itemField,
+   std::size_t arrayLength)
```

Comment:
UUIC, the `arrayLength` argument is not required, given that this information can be obtained from the number of repetitions of the child field `itemField`.

Reply:
It is not the way I understood it. `itemField` is the field representing the concrete data type held by the collection (in this case the collection is `std::array`). For example, with `std::array<int>` the `itemField` would represent the `int` thus would have no repetitions. I also tested removing this constructor argument and the tests would fail because of `fArrayLength` always being equal to 0 in case of simple collections as `std::array<int>`